SharpNoise/Modules/Discrete.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+
+namespace SharpNoise.Modules
+{
+    /// <summary>
+    /// This module take a value from a source module and from a given input "slice", return a discrete value, so its name, in the interval [i/slice, (i + 1)/slice]
+    /// For example, if slice count is set to 4, the Perlin source module values are converted into 0, 1/4, 2/4, 3/4 and 1 (4/4) ONLY, creating a sharp map which can be use
+    /// as filter for a select module or as-is for a tile map based game.
+    /// </summary>
+    /// <remarks>
+    /// This noise module requires one source module.
+    /// </remarks>
+    [Serializable]
+    public class Discrete : Module
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public Discrete()
+            : base(1)
+        {
+
+        }
+
+        /// <summary>
+        /// Gets or sets the number of maximum values to get from
+        /// input value.
+        /// </summary>
+        public UInt16 Values { get; set; }

```

Comment:
Unsigned integers are not CLS compliant. Now, it may well be that SharpNoise is not CLS compliant at the moment, but I would still not add any non-compliant features unless there's a real reason.